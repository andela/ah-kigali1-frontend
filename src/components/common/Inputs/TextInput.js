import React from "react";
import PropTypes from "prop-types";

const TextInput = ({
  name,
  type,
  id,
  placeholder,
  value,
  onChange,
  ...props
}) => (
  <input
    {...props}
    name={name}
    type={type}
    id={id}
    placeholder={placeholder}
    value={value}
    onChange={onChange}
  />
);

TextInput.propTypes = {
  name: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  id: PropTypes.string,
  placeholder: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  value: PropTypes.string,
  onKeyDown: PropTypes.func
};

TextInput.defaultProps = {
  id: "",
  value: PropTypes.string,
  onKeyDown: PropTypes.func
};

TextInput.defaultProps = {
  id: "",
  value: PropTypes.string,
  onKeyDown: PropTypes.func
};

TextInput.defaultProps = {
  id: "",
  value: "",
  onKeyDown: () => ""
};

export default TextInput;
