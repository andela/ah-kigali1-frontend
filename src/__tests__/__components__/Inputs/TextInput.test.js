import React from "react";
import { shallow } from "enzyme";
import TextInput from "../../../components/common/Inputs/TextInput";
const props = {
  type: "text",
  placeholder: "Email",
  onChange: jest.fn(),
  value: ""
};
describe("Login component", () => {
  let component;
  beforeEach(() => {
    component = shallow(<TextInput {...props} />);
  });
  test("should have input tag for email with initial state", () => {
    expect(component.props().value).toEqual("");
    expect(component.props().placeholder).toEqual("Email");
    component.simulate("change", {
      target: {
        name: "email",
        value: "luc.bayo@gmail.com"
      }
    });
    expect(component.props().onChange).toHaveBeenCalled();
  });
});
