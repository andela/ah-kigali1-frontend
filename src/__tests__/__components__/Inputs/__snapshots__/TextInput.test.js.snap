// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component should match the snapshot 1`] = `
<input
  id=""
  name="email"
  onChange={[MockFunction]}
  placeholder="Email"
  type="text"
  value=""
/>
`;
